src/handlers/grep.rs

```diff
@@ -222,14 +222,14 @@ fn get_code_style_sections<'b>(
     non_match_style: Style,
     grep: &GrepLine,
 ) -> Option<StyleSectionSpecifier<'b>> {
-    if let Some(raw_code_start) = ansi::ansi_preserving_index(
+    if let Some(prefix_end) = ansi::ansi_preserving_index(
         raw_line,
         match grep.line_number {
-            Some(n) => format!("{}:{}:", grep.path, n).len(),
-            None => grep.path.len() + 1,
+            Some(n) => format!("{}:{}:", grep.path, n).len() - 1,
+            None => grep.path.len(),
         },
     ) {
-        let match_style_sections = ansi::parse_style_sections(&raw_line[raw_code_start..])
+        let match_style_sections = ansi::parse_style_sections(&raw_line[(prefix_end + 1)..])
```

Comment:
If there are no escape-code-sequences separating the prefix-section from the code-section, the old approach would have produced the same value as the new approach. If there _are_ escape-code-sequences between the two, the old approach would have skipped them, which isn't what we want. So I _think_ this is okay?